### ISSUE TITLE
Ratelimiting user logins

### ISSUE THREAD

**Issue by lunny** [go-gitea/gitea#21906]
### Feature Description

When a user inputs wrong password several times, Gitea should lock x minutes from that session to login again.

**Comment by silverwind** [comment]
I suggest a high default like 10 logins in a 5 minute window because nothing is more obnoxious than when you're trying a few passwords that the site rate limits the retries.

**Comment by techknowlogick** [comment]
@silverwind yeah, IP because if you do it by username you could essentially lock out  users by failing to log in as them a few times.

**Comment by techknowlogick** [comment]
I have a very rough PR for this here: https://github.com/go-gitea/gitea/pull/21181

**Comment by silverwind** [comment]
I totally forgot about that one 😉

**Comment by eloydegen** [comment]
A CAPTCHA would be the best method in my view, otherwise there is still a risk of getting locked out by an attacker when something like Carrier Grade NAT is involved, although I'd see that as low risk.

**Comment by lunny** [comment]
Close since login with CAPTCHA has been implemented.

**Comment by techknowlogick** [comment]
CAPTCHAs aren't accessible, and so ratelimiting could provide an alternative

**Comment by silverwind** [comment]
We can offer both options, yes.
